e2e/help/help.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package help
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sylabs/singularity/internal/pkg/test"
+	"gotest.tools/assert"
+	"gotest.tools/golden"
+)
+
+type testingEnv struct {
+	// base env for running tests
+	CmdPath     string `split_words:"true"`
+	TestDir     string `split_words:"true"`
+	RunDisabled bool   `default:"false"`
+}
+
+var testenv testingEnv
+
+var helpTests = []struct {
+	cmds []string
+}{
+	// singularity oci
+	{[]string{"oci"}},
+	{[]string{"oci", "attach"}},
+	{[]string{"oci", "create"}},
+	{[]string{"oci", "delete"}},
+	{[]string{"oci", "exec"}},
+	{[]string{"oci", "kill"}},
+	{[]string{"oci", "mount"}},
+	{[]string{"oci", "pause"}},
+	{[]string{"oci", "resume"}},
+	{[]string{"oci", "run"}},
+	{[]string{"oci", "start"}},
+	{[]string{"oci", "state"}},
+	{[]string{"oci", "umount"}},
+	{[]string{"oci", "update"}},
+}
+
+func TestHelp(t *testing.T) {
+	c := test.NewCmd("singularity")
+
+	for _, tc := range helpTests {
+		name := fmt.Sprintf("%s.txt", strings.Join(tc.cmds, "-"))
+		path := filepath.Join("help", name)
+		args := append([]string{"help"}, tc.cmds...)
+
+		t.Run(name, func(t *testing.T) {
+			got := c.Run(t, args...).Stdout()
```

Comment:
This seems like it would add a lot of extra instances (1 per loop?) of the struct to be garbage collected when just one instance seems like enough. What do you think @mem ?